Cython/Utils.py

```diff
@@ -84,6 +85,34 @@ def file_newer_than(path, time):
     ftime = modification_time(path)
     return ftime > time
 
+
+def safe_makedirs(path):
+    try:
+        os.makedirs(path)
+    except OSError:
+        if not os.path.isdir(path):
+            raise
+
+
+def copy_file_to_dir_if_changed(sourcefile, destdir):
+    """
+    Copy file sourcefile to directory destdir (creating it if needed),
+    preserving metadata. If the destination file exists and is not
+    older than the source file, the copying is skipped.
+    """
+    destfile = os.path.join(destdir, os.path.basename(sourcefile))
+    try:
+        desttime = modification_time(destfile)
+    except OSError:
+        # New file does not exist, destdir may or may not exist
+        safe_makedirs(destdir)
+    else:
+        # New file already exists
+        if not file_newer_than(sourcefile, desttime):
```

Comment:
> This seems a bit risky to me--how can we be sure this is not an edited file that should be overwritten?

By the way, isn't this how dependency checking always works? You check whether the source file is newer than the target. It's what distutils does, so why should this particular case be different?